lib/profile.js

```diff
@@ -144,83 +154,87 @@ const writableProfileKeys = [
   'github',
 ]
 
-function set (args) {
-  const conf = npm.flatOptions
+async function set (args) {
+  const conf = { ...npm.flatOptions }
   const prop = (args[0] || '').toLowerCase().trim()
+
   let value = args.length > 1 ? args.slice(1).join(' ') : null
+
+  const readPasswords = async () => {
+    const newpassword = await readUserInfo.password('New password: ')
+    const confirmedpassword = await readUserInfo.password('       Again:     ')
+
+    if (newpassword !== confirmedpassword) {
+      log.warn('profile', 'Passwords do not match, please try again.')
+      return await readPasswords()
```

Comment:
return await is redundant here, all things returned from an async fucntuon are already awaited
```suggestion
      return readPasswords()
```

Reply:
****